### ISSUE TITLE
Option for using only dash variants

### ISSUE THREAD

**Issue by rggjan** [lebrice/SimpleParsing#67]
There is an option `add_option_string_dash_variants`:

```
from dataclasses import dataclass

from simple_parsing import ArgumentParser


@dataclass
class Args:
    enable_this_option: bool  # This is a nice option
    num_spiders_per_web: int = 5  # How many?


parser = ArgumentParser(add_option_string_dash_variants=True)
parser.add_arguments(Args, dest="args")

args = parser.parse_args(["--help"])
```

However, this always creates two versions of the same option and clutters the help:
```
  --enable-this-option bool, --enable_this_option bool
                        This is a nice option (default: None)
  --num-spiders-per-web int, --num_spiders_per_web int
                        How many? (default: 5)
```

It would be nice to have the option to disable the underscore variants and only use the dash variants (for people preferring dashes):
```
  --enable-this-option bool
                        This is a nice option (default: None)
  --num-spiders-per-web int
                        How many? (default: 5)
```

**Comment by lebrice** [comment]
Hey again @rggjan, thanks for posting!

Sounds good! And would also be pretty easy to add, so I'll look into this soon. I'll perhaps end up adding another Enum for the option for that, a bit like for `ConflictResolution`.

**Comment by lebrice** [comment]
Hey @rggjan, Sorry it's been a while. 

I'd really appreciate someone making a PR for this, since the change would be pretty simple (and quite similar to what I describe in #75):
> - First, take a look at [how the `FieldWrapper` currently creates the option strings:](https://github.com/lebrice/SimpleParsing/blob/7cf0ed5456827fa844790085d46db864af73a9d2/simple_parsing/wrappers/field_wrapper.py#L468)
> - then, take a look at [how they are used in the `DataclassWrapper` to actually create the arguments:](https://github.com/lebrice/SimpleParsing/blob/master/simple_parsing/wrappers/dataclass_wrapper.py#L103)
>
> You'd need to either change what value is returned by the `option_strings` property so it only includes the full path (the `dest` attribute on the `FieldWrapper`), or filter out the option strings that are passed to the `add_argument` function.

In this case, there could also be some kind of enum that gets passed to the `ArgumentParser` constructor, that defines what style of option strings to use, something like:
```python3
class OptionStringStyle(enum.Enum):
     DEFAULT = 0
     WITH_DASHES = 1
     ONLY_DASHES = 2
```
Hope this helps! Let me know what you think!

**Comment by rggjan** [comment]
Thanks for the info and hints, makes sense! Not sure if / when I can take a look at this, though.

**Comment by lebrice** [comment]
Fixed by #93
